Installation issue

Hello,
I'm trying to install Geotrek following the documentation, and I have some problems.

At the very beginnig, when I run the install.sh, the script can't find the `etc/setting.ini` file. I checked, and I have no `'etc'` folder at all... So the install aborted. 
I tried to create myself this folder and the `settings.ini` file with the variable expected (dbhost, dbname etc...). It works (the database is installed), but the install crash few step later when it try to install the python environnement. `Could not setup python environment !`

Did I miss something in the installation documentation ?
How can I fix this problem ?

Thanks for your help

Hello,

I have an issue with the install of Geotrek (2.15.2). When I run ./install.sh (I follow the method described by @camillemonchicourt in her post of the 01242017) the install top with the following code : 

```
error: Setup script exited with error: command 'x86_64-linux-gnu-gcc' failed with exit status 1
An error occurred when trying to install GDAL 1.10.0. Look above this message for any errors that were output by easy_install.
While:
  Installing gdal-bindings.
Error: Couldn't install: GDAL 1.10.0
Makefile:70 : la recette pour la cible « env_standalone » a échouée
make: *** [env_standalone] Erreur 1
+ success=2
+ '[' 2 -ne 0 ']'
+ exit_error 3 'Could not setup python environment !'
+ code=3
+ shift
+ echo_error 'Could not setup python environment !'
+ set +x
+ echo '(More details in install.log)'
(More details in install.log)
+ exit 3
```

I tried to install many libs of Gdal and Python but it doesn't works.

If anyone have an advice for me ?

Pierre

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,7 +1,5 @@
 language: python
 
-dist: precise
-
 python:
   - "2.7"
 
@@ -10,6 +8,7 @@ env:
   - ACTION=deploy
   - ACTION=integration
   - ACTION=codestyle
+
 
 matrix:
   allow_failures:
@@ -21,37 +20,26 @@ services:
 cache:
   - apt
 
+dist: trusty
+sudo: required
+
 before_install:
+  # Fix postgresql / postgis for travis trusty
+  - echo $PATH
+  - sudo apt-get update -qq
+  - sudo apt-get install -y -qq unoconv inkscape
+  - sudo apt-get remove postgis*
+  - sudo apt-get remove postgresql*
+  - sudo rm -rf /usr/local/bin/phantomjs*
+  - sudo rm -rf /usr/bin/phantomjs*
+  - sudo rm -rf /usr/local/phantomjs*
+
   # Download bundle of python eggs, and use it as download cache
   - mkdir -p $HOME/.buildout/downloads
   - echo -e "[buildout]\ndownload-cache=$HOME/.buildout/downloads" > $HOME/.buildout/default.cfg
   - "wget http://prod-geotrek-fr.makina-corpus.net/media/upload/geotrek-bundle.tar.gz -O /tmp/geotrek-bundle.tar.gz || true"
   - "tar -zxf /tmp/geotrek-bundle.tar.gz --directory=$HOME || true"
-
-  # GDAL-fix -- packaging is miserable, no way to obtain libgdal-dev for libgdal1 when libgdal1h available
-  - sudo apt-get update -qq
-  - sudo apt-get download -y libgdal-dev
-  - sudo dpkg-deb -x libgdal-dev*.deb /
-  - sudo ln -sf /usr/lib/libgdal.so.1 /usr/lib/libgdal.so
-  - sudo ldconfig
-  # GDAL-fix -- Do not install GDAL from install script
-  - sed -i "s/gdal-bin//" install.sh
-  - sed -i "s/libgdal-dev//" install.sh
-  - sed -i "s/GDAL=.*/GDAL=1.9.0/" conf/buildout.cfg
-  # GDAL-fix -- Do not install postgis locally, trick test on host
   - make etc/settings.ini
-  - sed -i "s/localhost/127.0.0.1/" etc/settings.ini
-  # GDAL-fix -- Use Travis PostGIS, initialize DB
-  - sudo sed -i "s/^client_min_messages.*$/client_min_messages = log/" /etc/postgresql/*/main/postgresql.conf
-  - sudo /etc/init.d/postgresql restart
-  - psql -c "CREATE DATABASE geotrekdb ENCODING 'UTF8' TEMPLATE template0;" -U postgres
-  - psql -c 'CREATE EXTENSION postgis;' -U postgres -d geotrekdb
-  - psql -c "CREATE USER geotrek WITH PASSWORD 'geotrek';" -U postgres
-  # django tests require to be superuser to create test db with postgis extension
-  - if [[ $ACTION == test ]]; then psql -c "ALTER USER geotrek SUPERUSER;" -U postgres; fi
-  - psql -c 'GRANT ALL PRIVILEGES ON DATABASE geotrekdb TO geotrek;' -U postgres
-  - psql -c 'GRANT ALL ON spatial_ref_sys,geometry_columns,raster_columns TO geotrek;' -U postgres -d geotrekdb
-  # GDAL-fix -- End (remove with Ubuntu 14.04)
 
 install:
   - if [[ $ACTION != codestyle ]]; then deactivate; fi
@@ -83,5 +71,5 @@ after_failure:
   - cat var/log/nginx-error.log
   - cat var/log/supervisord.log
 
-notifications:
-  irc: "irc.freenode.org#geotrek"
+#notifications:
+  #irc: "irc.freenode.org#geotrek"

--- a/.travis.yml
+++ b/.travis.yml
@@ -28,8 +28,10 @@ before_install:
   - echo $PATH
   - sudo apt-get update -qq
   - sudo apt-get install -y -qq unoconv inkscape
-  - sudo apt-get remove postgis*
-  - sudo apt-get remove postgresql*
+  # Make sure to use postgresql and postgis version from main Ubuntu repository
+  - sudo rm -f /etc/apt/sources.list.d/pgdg.list
+  - sudo apt-get remove -qq postgis*
+  - sudo apt-get remove -qq postgresql*
   - sudo rm -rf /usr/local/bin/phantomjs*
   - sudo rm -rf /usr/bin/phantomjs*
   - sudo rm -rf /usr/local/phantomjs*

--- a/.travis.yml
+++ b/.travis.yml
@@ -46,8 +46,8 @@ before_install:
 install:
   - if [[ $ACTION != codestyle ]]; then deactivate; fi
   - if [[ $ACTION == codestyle ]]; then pip install flake8; fi
-  - if [[ $ACTION == test ]]; then ./install.sh --tests || (cat ./install.log; false); fi
-  - if [[ $ACTION == deploy || $ACTION == integration ]]; then ./install.sh --noinput || (cat ./install.log; false); fi
+  - if [[ $ACTION == test ]]; then ./install.sh --tests --noinput; fi
+  - if [[ $ACTION == deploy || $ACTION == integration ]]; then ./install.sh --noinput; fi
 script:
   - if [[ $ACTION == codestyle ]]; then flake8 --exclude "" --ignore=E501,F403,F405 geotrek/settings; fi
   - if [[ $ACTION == codestyle ]]; then flake8 geotrek; fi
@@ -67,7 +67,6 @@ after_success:
   - if [[ $ACTION == test ]]; then coveralls; fi
 
 after_failure:
-  - cat install.log
   - cat var/log/geotrek.log
   - cat var/log/nginx-access.log
   - cat var/log/nginx-error.log

--- a/.travis.yml
+++ b/.travis.yml
@@ -68,9 +68,10 @@ after_success:
 
 after_failure:
   - cat var/log/geotrek.log
-  - cat var/log/nginx-access.log
-  - cat var/log/nginx-error.log
-  - cat var/log/supervisord.log
+  - sudo cat /var/log/nginx/geotrek_access.log
+  - sudo cat /var/log/nginx/geotrek_error.log
+  - cat /var/log/supervisord.log
+  - cat /var/log/geotrek.log
 
 #notifications:
   #irc: "irc.freenode.org#geotrek"

--- a/.travis.yml
+++ b/.travis.yml
@@ -35,6 +35,8 @@ before_install:
   - sudo rm -rf /usr/local/bin/phantomjs*
   - sudo rm -rf /usr/bin/phantomjs*
   - sudo rm -rf /usr/local/phantomjs*
+  # Make sure nginx has access to static files
+  - sudo chmod a+x /home/travis
 
   # Download bundle of python eggs, and use it as download cache
   - mkdir -p $HOME/.buildout/downloads

--- a/Makefile
+++ b/Makefile
@@ -9,10 +9,6 @@ version=$(shell git describe --tags --abbrev=0)
 
 ROOT_DIR=$(shell pwd)
 BUILDOUT_CFG = $(ROOT_DIR)/conf/buildout.cfg
-BUILDOUT_VERSION = 1.7.1
-BUILDOUT_BOOTSTRAP_URL = http://downloads.buildout.org/2/bootstrap.py
-BUILDOUT_BOOTSTRAP = bootstrap.py
-BUILDOUT_BOOTSTRAP_ARGS = -c $(BUILDOUT_CFG) --buildout-version=$(BUILDOUT_VERSION) buildout:directory=$(ROOT_DIR)
 BUILDOUT = bin/buildout
 BUILDOUT_ARGS = -N buildout:directory=$(ROOT_DIR) buildout:user=$(user)
 
@@ -27,12 +23,9 @@ etc/settings.ini:
 
 bin/python:
 	virtualenv -p /usr/bin/python2 .
-	bin/pip install -U setuptools
+	bin/pip install -U setuptools==38.7.0 zc.buildout==2.11.1
 	mkdir -p lib/src
 	mkdir -p lib/eggs
-	wget --quiet -O $(BUILDOUT_BOOTSTRAP) $(BUILDOUT_BOOTSTRAP_URL)
-	bin/python $(BUILDOUT_BOOTSTRAP) $(BUILDOUT_BOOTSTRAP_ARGS)
-	rm $(BUILDOUT_BOOTSTRAP)
 
 install: etc/settings.ini bin/python
 
@@ -115,11 +108,11 @@ deploy: update
 	sudo service supervisor restart
 
 all_makemessages:
-	for dir in `find geotrek/ -type d -name locale`; do pushd `dirname $$dir` > /dev/null; $(ROOT_DIR)/bin/django-admin makemessages --no-location --all; popd > /dev/null; done
+	for dir in `find geotrek/ -type d -name locale`; do pushd `dirname $$dir` > /dev/null; bin/django makemessages --no-location --all; popd > /dev/null; done
 
 all_compilemessages:
-	for dir in `find geotrek/ -type d -name locale`; do pushd `dirname $$dir` > /dev/null; $(ROOT_DIR)/bin/django-admin compilemessages; popd > /dev/null; done
-	for dir in `find lib/src/ -type d -name locale`; do pushd `dirname $$dir` > /dev/null; $(ROOT_DIR)/bin/django-admin compilemessages; popd > /dev/null; done
+	for dir in `find geotrek/ -type d -name locale`; do pushd `dirname $$dir` > /dev/null; bin/django compilemessages; popd > /dev/null; done
+	for dir in `find lib/src/ -type d -name locale`; do pushd `dirname $$dir` > /dev/null; bin/django compilemessages; popd > /dev/null; done
 
 
 

--- a/Makefile
+++ b/Makefile
@@ -85,7 +85,7 @@ node_modules:
 	npm install geotrek/jstests
 
 test_js: node_modules
-	./node_modules/geotrek-tests/node_modules/mocha-phantomjs/bin/mocha-phantomjs geotrek/jstests/index.html
+	./node_modules/.bin/mocha-phantomjs geotrek/jstests/index.html
 
 tests: test test_js test_nav
 

--- a/conf/buildout.cfg
+++ b/conf/buildout.cfg
@@ -26,7 +26,6 @@ eggs =
     geotrek
 auto-checkout =
     django-modeltranslation
-    isotoma.recipe.django
     screamshotter
     pygal
     django-leaflet
@@ -39,7 +38,7 @@ include-dirs = /usr/include/gdal
 library-dirs = /usr/lib
 
 [django]
-recipe = isotoma.recipe.django
+recipe = djangorecipe
 project = geotrek
 wsgi = true
 eggs =
@@ -82,7 +81,7 @@ eggs =
 # Deployment tools
 #
 mr.developer = 1.25
-isotoma.recipe.django = 3.1.7
+djangorecipe = 2.2.1
 hexagonit.recipe.download = 1.5.1
 z3c.recipe.mkdir = 0.3.1
 zc.recipe.egg = 1.3.2
@@ -147,7 +146,6 @@ idna = 2.5
 #
 # Not (yet) released on Pypi
 django-modeltranslation = git https://github.com/makinacorpus/django-modeltranslation.git branch=0.12.1
-isotoma.recipe.django = git https://github.com/isotoma/isotoma.recipe.django.git rev=bb285b016c7f3ea470c6522acd7fcec97983c0b2
 screamshotter = git https://github.com/makinacorpus/django-screamshot.git branch=screamshotter
 # Custom
 pygal = git https://github.com/makinacorpus/pygal.git branch=master

--- a/docs/changelog.rst
+++ b/docs/changelog.rst
@@ -5,6 +5,11 @@ CHANGELOG
 2.17.3.dev0
 -----------
 
+**Breaking changes**
+
+* Drop support of Ubuntu Precise 12.04, please upgrade to
+  Trusty 14.04 or Xenial 16.04 before upgrading Geotrek-admin
+
 **Preventive maintainance**
 
 * Upgrade to Django 1.9

--- a/geotrek/jstests/index.html
+++ b/geotrek/jstests/index.html
@@ -7,9 +7,9 @@
 <body>
   <div id="map" style="display: none; height: 300px"></div>
   <div id="mocha"></div>
-  <script src="../../node_modules/geotrek-tests/node_modules/chai/chai.js"></script>
-  <script src="../../node_modules/geotrek-tests/node_modules/mocha/mocha.js"></script>
-  <script src="../../node_modules/geotrek-tests/node_modules/leaflet/debug/leaflet-include.js"></script>
+  <script src="../../node_modules/chai/chai.js"></script>
+  <script src="../../node_modules/mocha/mocha.js"></script>
+  <script src="../../node_modules/leaflet/debug/leaflet-include.js"></script>
 
   <!-- Use Javascript libs bundled in mapentity -->
   <script src="../../lib/parts/omelette/mapentity/static/mapentity/jquery.min.js"></script>

--- a/geotrek/zoning/factories.py
+++ b/geotrek/zoning/factories.py
@@ -11,10 +11,13 @@ from geotrek.core.factories import TopologyFactory
 from . import models
 
 
+# Don't intersect with geom from PathFactory
+SPATIAL_EXTENT = (200000, 300000, 1100000, 1200000)
+
 # Create 16 cities and 4 districts distinct same-area zone covering the spatial_extent and cycle on it
-geom_city_iter = bbox_split_srid_2154(settings.SPATIAL_EXTENT, by_x=4, by_y=4, cycle=True)
-geom_district_iter = bbox_split_srid_2154(settings.SPATIAL_EXTENT, by_x=2, by_y=2, cycle=True)
-geom_area_iter = bbox_split_srid_2154(settings.SPATIAL_EXTENT, by_x=2, by_y=2, cycle=True)
+geom_city_iter = bbox_split_srid_2154(SPATIAL_EXTENT, by_x=4, by_y=4, cycle=True)
+geom_district_iter = bbox_split_srid_2154(SPATIAL_EXTENT, by_x=2, by_y=2, cycle=True)
+geom_area_iter = bbox_split_srid_2154(SPATIAL_EXTENT, by_x=2, by_y=2, cycle=True)
 
 
 class CityFactory(factory.DjangoModelFactory):

--- a/install.sh
+++ b/install.sh
@@ -203,20 +203,13 @@ function minimum_system_dependencies {
     echo_progress
     sudo apt-get install -y -qq python unzip wget python-software-properties
     echo_progress
-    if [ $precise -eq 1 ]; then
-        sudo apt-add-repository -y ppa:git-core/ppa
-        sudo apt-add-repository -y ppa:ubuntugis/ppa
-        sudo apt-get update -qq
-        echo_progress
-    fi
-
     sudo apt-get install -y -qq git gettext python-virtualenv build-essential python-dev
     echo_progress
 }
 
 
 function geotrek_system_dependencies {
-    sudo apt-get install -y -q --no-upgrade libjson0 gdal-bin libgdal-dev libssl-dev
+    sudo apt-get install -y -q --no-upgrade libjson0 gdal-bin libgdal-dev libssl-dev binutils libproj-dev
     
     if [ $xenial -eq 1 ]; then
         sudo apt-get install libgeos-c1v5 libproj9
@@ -527,10 +520,14 @@ function geotrek_setup {
         echo_step "Generate services configuration files..."
 
         # restart supervisor in case of xenial before 'make deploy'
-        sudo service supervisor force-stop && sudo service supervisor stop && sudo service supervisor start
+        if [ $trusty -eq 1 ]; then
+            sudo service supervisor force-stop && sudo service supervisor stop && sudo service supervisor start
+        fi
         if [ $? -ne 0 ]; then
             exit_error 10 "Could not restart supervisor !"
         fi
+
+        echo_progress
 
         # If buildout was successful, deploy really !
         if [ -f /etc/supervisor/supervisord.conf ]; then
@@ -598,10 +595,7 @@ trusty=$(grep "Ubuntu 14.04" /etc/issue | wc -l)
 vivid=$(grep "Ubuntu 15.04" /etc/issue | wc -l)
 xenial=$(grep "Ubuntu 16.04" /etc/issue | wc -l)
 
-if [ $precise -eq 1 ]; then
-    psql_version=9.1
-    pgis_version=2.0
-elif [ $trusty -eq 1 ]; then
+if [ $trusty -eq 1 ]; then
     psql_version=9.3
     pgis_version=2.1
 elif [ $vivid -eq 1 ]; then
@@ -612,8 +606,10 @@ elif [ $xenial -eq 1 ]; then
     pgis_version=2.2
 fi
 
-if [ $precise -eq 1 -o $trusty -eq 1 -o $vivid -eq 1 -o $xenial -eq 1 ] ; then
+if [ $trusty -eq 1 -o $vivid -eq 1 -o $xenial -eq 1 ] ; then
     geotrek_setup
+elif [ $precise -eq 1 ] ; then
+    exit_error 5 "Support for Ubuntu Precise 12.04 was dropped. Upgrade your server first. Aborted."
 else
     exit_error 5 "Unsupported operating system. Aborted."
 fi

--- a/install.sh
+++ b/install.sh
@@ -9,16 +9,6 @@ cd "$(dirname "$0")"
 
 #------------------------------------------------------------------------------
 
-# Redirect whole output to log file
-rm -f install.log
-touch install.log
-chmod 600 install.log
-
-exec 3>&1 4>&2
-exec 1> install.log 2>&1
-
-#------------------------------------------------------------------------------
-
 STABLE_VERSION=${STABLE_VERSION:-2.17.2}
 dev=false
 tests=false
@@ -29,7 +19,6 @@ settingsfile=etc/settings.ini
 
 
 usage () {
-    exec 2>&4
     cat >&2 <<- _EOF_
 Usage: Install project [OPTIONS]
     -d, --dev         minimum dependencies for development
@@ -39,7 +28,6 @@ Usage: Install project [OPTIONS]
     -s, --standalone  deploy a single-server production instance (Default)
     -h, --help        show this help
 _EOF_
-    exec 2>&1
     return
 }
 
@@ -67,6 +55,17 @@ while [[ -n $1 ]]; do
     shift
 done
 
+#------------------------------------------------------------------------------
+
+# Redirect whole output to log file
+if $interactive ; then
+    rm -f install.log
+    touch install.log
+    chmod 600 install.log
+
+    exec 3>&1 4>&2
+    exec 1> install.log 2>&1
+fi
 
 #------------------------------------------------------------------------------
 #
@@ -76,36 +75,36 @@ done
 
 function echo_step () {
     set +x
-    exec 2>&4
+    if $interactive; then exec 2>&4; fi
     echo -e "\n\e[92m\e[1m$1\e[0m" >&2
-    exec 2>&1
+    if $interactive; then exec 2>&1; fi
     set -x
 }
 
 
 function echo_warn () {
     set +x
-    exec 2>&4
+    if $interactive; then exec 2>&4; fi
     echo -e "\e[93m\e[1m$1\e[0m" >&2
-    exec 2>&1
+    if $interactive; then exec 2>&1; fi
     set -x
 }
 
 
 function echo_error () {
     set +x
-    exec 2>&4
+    if $interactive; then exec 2>&4; fi
     echo -e "\e[91m\e[1m$1\e[0m" >&2
-    exec 2>&1
+    if $interactive; then exec 2>&1; fi
     set -x
 }
 
 
 function echo_progress () {
     set +x
-    exec 2>&4
+    if $interactive; then exec 2>&4; fi
     echo -e ".\c" >&2
-    exec 2>&1
+    if $interactive; then exec 2>&1; fi
     set -x
 }
 
@@ -120,9 +119,11 @@ function exit_error () {
 
 function echo_header () {
     set +x
-    exec 2>&4
-    cat docs/logo.ans >&2
-    exec 2>&1
+    if $interactive; then
+        exec 2>&4
+        cat docs/logo.ans >&2
+        exec 2>&1
+    fi
     set -x
     version=$(cat VERSION)
     echo_step      "... install $version" >&2

--- a/install.sh
+++ b/install.sh
@@ -428,6 +428,10 @@ function geotrek_setup {
 
     # Python bootstrap
     make install
+    success=$?
+    if [ $success -ne 0 ]; then
+        exit_error 2 "Could not setup virtualenv/buildout !"
+    fi
     echo_progress
 
     if $freshinstall && $interactive && ($prod || $standalone) ; then

